mountaineer/paths.py

```diff
@@ -146,19 +147,34 @@ def copy(self) -> "ManagedViewPath":
         path.package_root_link = self.package_root_link
         return path
 
-    def rglob(self, pattern: str):
-        # Override the rglob method to return a ManagedViewPath
-        for path in super().rglob(pattern):
-            yield self._inherit_root_link(path)
+    if sys.version_info >= (3, 12):
```

Comment:
Address the full new implementation surface if users are running above 3.12. We don't currently use any of these features, but this makes pyright happy and might be leveraged in the future.